jdaviz/configs/imviz/tests/test_delete_data.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from astropy.io import fits
+from numpy.testing import assert_allclose
+from regions import PixCoord, CirclePixelRegion, RectanglePixelRegion
+
+from jdaviz.configs.imviz.tests.utils import BaseImviz_WCS_WCS
+
+
+class TestDeleteData(BaseImviz_WCS_WCS):
+
+    def test_delete_with_subset_wcs(self):
+        # Add a third dataset to test relinking
+        arr = np.ones((10, 10))
+
+        # First data with WCS, same as the one in BaseImviz_WCS_NoWCS.
+        hdu3 = fits.ImageHDU(arr, name='SCI')
+        hdu3.header.update({'CTYPE1': 'RA---TAN',
+                            'CUNIT1': 'deg',
+                            'CDELT1': -0.0002777777778,
+                            'CRPIX1': 1,
+                            'CRVAL1': 337.5202808,
+                            'NAXIS1': 10,
+                            'CTYPE2': 'DEC--TAN',
+                            'CUNIT2': 'deg',
+                            'CDELT2': 0.0002777777778,
+                            'CRPIX2': 1,
+                            'CRVAL2': -20.833333059999998,
+                            'NAXIS2': 10})
```

Comment:
I think all you need is this. See the actual baseclass implementation in `imviz/tests/utils.py`.
```suggestion
        hdu3 = NDData(arr, wcs=self.wcs_1)
```